### ISSUE TITLE
How to download FULL video in youtube-dl?

### ISSUE THREAD

**Issue by HVEC4** [rg3/youtube-dl#14784]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.11.15*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [X] I've **verified** and **I assure** that I'm running youtube-dl **2017.11.15**

### Before submitting an *issue* make sure you have:
- [X] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [X] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [X] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### Description of your *issue*, suggested solution and other information

Explanation of your *issue* in arbitrary form goes here. Please make sure the [description is worded well enough to be understood](https://github.com/rg3/youtube-dl#is-the-description-of-the-issue-itself-sufficient). Provide as much context and examples as possible.
If work on your *issue* requires account credentials please provide them or explain how one can obtain them.

Hello! I just got youtube-dl and ffmpeg today, and I've actually come up with the opposite issue that I've seen everywhere in all my searches. I am trying to figure out how to download a FULL youtube video with youtube-dl. Everything I'm seeing looks to be like everyone is already able to do this and is trying to figure out how to just download tidbits. I am having the opposite issue.

Like everyone else, the highest merged quality available for download is 720p on my 1080p video, but separate, the video and audio files are 1080p. I enter in -f 137+140 "link" and they download and merge when finished, but the video will freeze after 5 seconds, every single time I try this. The audio is always fine. At one point I tried again and the audio download actually didn't go through, so I took that opportunity to check on the video file to see what was wrong. Sure enough, the video itself is only being downloaded 5 seconds long. This original video is over 12 minutes, and I am trying to get the whole thing. I have NO idea why only the first 5 seconds are being downloaded. The time it takes and the file size indicate it is the full video, but it is only 5 seconds long. I deleted that, yet again, and then downloaded the audio by itself again, which is fine. I'm now in the process of downloading the video 100% by itself to see if that does anything, but it looks the exact same in my terminal as it has the last 20 times, same size, same time estimated, everything. Why is it only downloading 5 seconds of a 12 minute video?

**Comment by HVEC4** [comment]
Something I've since done is added "-v" for the output, so my command is now: [ youtube-dl -f 137+140 -v "link" ], and this actually got the video past 5 seconds! However, still only 5 minutes and 45 seconds, out of 12 minutes and 35 seconds. I have literally no clue what I can do. What I'm trying right this second is combining that with the ffmpeg timeframe command, so: [ ffmpeg -ss 00:00:00 -i $(youtube-dl -f 137+140 -v "link") -t 00:12:35 -c:v copy -c:a copy "title".mp4 ], and the "-v" part works immediately, the output data pops up, and this is the second time I'm trying this with the ffmpeg command, and it acts like it is doing something in my terminal, like if I go to exit out, it says it's busy and asks if I'm sure I want to quit, but nothing is actively coming up in the terminal. I *hope* it's just that the video is long so it's taking a minute, but nothing but the "-v" output logs have come up and the last time I tried this a little while ago, without "-v", I let it sit here for over an hour and gave up, with nothing popping up in terminal. It was thinking, but nothing happened in an hour.

And I'll also add that https is enabled in my ffmpeg configuration, but it isn't working for some reason. So a couple people suggested using youtube-dl's "--get-url" command and retrieving the direct urls for the audio and video files separately, then putting them into that long ffmpeg timeframe command ([ -i "top link" -i "bottom link" ]), but my terminal keeps coming up with "https error 404 not found" and when I change it to http, like keep the urls the same except take the s out, I get "http error 403 access denied" or something along those lines. So neither versions of the "get url" long links work for me, and I cannot download the entire video simply through youtube-dl and the normal video link, so I am trying the ffmpeg timeframe command with the normal link because the proper long ones won't work. I can leave it sitting here all night I guess, but if that doesn't work, I'm not sure what will. No clue why it went from 5 seconds to 5 minutes and 45 seconds and cannot, for the love of god, just download the whole thing. No one else seems to be having this issue.

PLEASE someone respond.

**Comment by HVEC4** [comment]
ALRIGHT IT FINALLY WORKED!! I let it sit and do its own thing, it never said anything more than the output that I forced it to give (with command "-v"), but I just left it alone, and it finished, and the whole video is there, 12:35 and 1080p! God bless

IF ANYONE HAPPENS TO HAVE THIS ISSUE....I doubt anyone will, I went years back on several threads trying to find this, but just in case, you will need to manually instruct it to download the entire video. The regular old youtube link will work, and the command is as follows (you may not need the output, I used it because at one point trying this or something similar I was given an "input/output" issue, so I'd just add it in, can't hurt):

`ffmpeg -ss 00:00:00 -i $(youtube-dl -f 137+140 -v "link") -t (duration in hh:mm:ss) -c:v copy -c:a copy "title".mp4`

This is the same command you use to download a specific part of the video. You can start and finish at any point, and you can use seconds to record time, or the hh:mm:ss format, which I prefer personally. The "-t hh:mm:ss" part is duration of time spent on the video, not the actual time mark of the video at which you want to stop downloading. Also, the file won't actually be renamed upon finishing, so I'm not sure what that needs to be there for, but that's the command I found and I didn't want to risk screwing it up lol.

Anyways, I'm closing this now, but it was....fun...screwing around with this all day. I now know for future videos haha

**Comment by Hrxn** [comment]
Wait..

> [..] but the video will freeze after 5 seconds [..]

You mean during playback? So, this is not an issue with the download at all?

**Comment by HVEC4** [comment]
Yes, the original video itself was fine, all things with the downloading process seemed to be fine (the completed video that I finally got was the SAME SIZE as the 5 second version so I have no idea what that was all about but it was going through the motions properly apparently), but the final product gave me perfect audio and crystal clear, 5-second-long video. The first few times I did it the video would just freeze after 5 seconds and remain on the same frame for the rest of the duration that the audio continued, and I only really saw what was happening when I opened the video and audio separately. The video was only being downloaded 5 seconds long. No idea why.

When I first got the input/output issue, I’m not sure what I did to bring that up, but after that I manually instructed it to list the output info, and then the 5 seconds downloaded was increased to almost 6 minutes. No clue what happened here either. I’m 100% new to this program, first day using it, trial and error lol.

So I coupled that with manually telling it to download the entire duration, and that’s what finally worked. No issues with original audio or video, and it *appeared* to be downloading just fine, but I’d find out every single time afterward, however, that it was for some reason not. No idea why. Could not find duplicates anywhere online about this for years back lol
